semgrep-core/src/analyzing/AST_to_IL.ml

```diff
@@ -753,7 +765,40 @@ let rec stmt_aux env st =
         List.map (stmt env) (st2 |> Common.opt_to_list) |> List.flatten
       in
       ss @ [ mk_s (If (tok, e', st1, st2)) ]
-  | G.Switch (_, _, _) -> todo (G.S st)
+  | G.Switch (tok, scrut_opt, cases_and_bodies) ->
+      let ss, translate_cases =
+        match scrut_opt with
+        | Some scrut ->
+            let ss, scrut' = expr_with_pre_stmts env scrut in
+            (ss, scrut_and_cases_to_exp env tok scrut scrut')
+        | None -> ([], cases_to_exp env tok)
+      in
+      let break_label, break_label_s, switch_env =
+        mk_switch_break_label env tok
+      in
+      let rec cases_and_bodies_to_stmts = function
```

Comment:
`stmt_aux` is already part of a group of mutually recursive functions, so I would avoid nesting another recursive function in here. We could put it as the same level as `python_with_stmt`.

Reply:
I guess it's because you want to avoid passing `translate_cases` to it ?